atlasdb-dbkvs/src/main/java/com/palantir/atlasdb/keyvalue/dbkvs/OracleTableNameGetter.java

```diff
@@ -67,6 +77,44 @@ public String getInternalShortOverflowTableName(ConnectionSupplier connectionSup
         return getPrefixedOverflowTableName(tableRef);
     }
 
+    public Set<TableReference> getTableReferencesFromShortTableNames(
+            ConnectionSupplier connectionSupplier, Set<String> shortTableNames) throws TableMappingNotFoundException {
+        return getTableReferencesFromShortTableNamesWithPrefix(connectionSupplier, shortTableNames, tablePrefix);
+    }
+
+    public Set<TableReference> getTableReferencesFromShortOverflowTableNames(
+            ConnectionSupplier connectionSupplier, Set<String> shortTableNames) throws TableMappingNotFoundException {
+        return getTableReferencesFromShortTableNamesWithPrefix(
+                connectionSupplier, shortTableNames, overflowTablePrefix);
+    }
+
+    private Set<TableReference> getTableReferencesFromShortTableNamesWithPrefix(
+            ConnectionSupplier connectionSupplier, Set<String> shortTableNames, String tablePrefixToStrip)
+            throws TableMappingNotFoundException {
+        Set<String> longTableNames = getLongTableNamesFromTableNames(connectionSupplier, shortTableNames);
+        return longTableNames.stream()
+                .peek(tableName -> {
+                    if (!StringUtils.startsWithIgnoreCase(tableName, tablePrefixToStrip)) {
+                        throw new SafeIllegalArgumentException(
+                                "Long table name does not begin with prefix",
+                                SafeArg.of("tableName", tableName),
```

Comment:
It doesn't seem obvious to me - how do we know that this is safe?

Reply:
This was straight up me being stupid, at 11:30pm or whenever I wrote this it was yeah atlasdb-proxy users declare these names woo safe! Despite telling our new hire yesterday that these names are in fact not safe.